File: app/src/main/java/com/monster/app/niceweather/base/BaseView.java
package com.monster.app.niceweather.base;

import android.content.Context;
import android.support.annotation.LayoutRes;
import android.widget.FrameLayout;

import butterknife.ButterKnife;

/**
 * Created by monster on 2/25/18.
 */

public class BaseView extends FrameLayout {

    public BaseView(Context context) {
        super(context);
    }

    public void inflate(@LayoutRes int layoutId) {
        inflate(getContext(), layoutId, this);
        ButterKnife.bind(this);
    }
}

File: app/src/main/java/com/monster/app/niceweather/base/BaseView.kt
package com.monster.app.niceweather.base

import android.content.Context
import android.support.annotation.LayoutRes
import android.view.View
import android.widget.FrameLayout

/**
 * Created by monster on 2/25/18.
 */

open class BaseView(context: Context) : FrameLayout(context) {

    fun inflate(@LayoutRes layoutId: Int) {
        View.inflate(context, layoutId, this)
    }
}
